lightning/src/ln/peer_handler.rs

```diff
@@ -670,6 +715,24 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 		}
 	}
 
+	///
+	pub fn enqueue_message_check_peer<M: Encode + Writeable + Debug>(&self, node_id: &PublicKey, message: &M) {
+		let mut peers_lock = self.peers.lock().unwrap();
+		let peers = &mut *peers_lock;
+		match peers.node_id_to_descriptor.get(node_id) {
+			Some(descriptor) => match peers.peers.get_mut(&descriptor) {
+				Some(mut peer) => {
+					if peer.their_features.is_some() {
```

Comment:
Maybe you could pass a predicate to this new method to evaluate if the peer not only identify to the given public key but also announce the feature signaling support for the relayed messages ?

Reply:
IIUC this is just to check that the init message was received before we try to send something. I kind of see what you mean I think but I feel better to keep it simple for now and maybe add this kind of possibilities after.